packages/sync/build.gradle.kts

```diff
@@ -0,0 +1,303 @@
+                implementation("androidx.test:rules:${Versions.androidxTest}")
+                implementation(kotlin("reflect:${Versions.kotlin}"))
+            }
+        }
+    }
+}
+
+kotlin {
+    sourceSets {
+        create("darwinCommon") {
+            dependsOn(getByName("commonMain"))
+        }
+    }
+}
+
+// IOS Configuration
+kotlin {
+    // For ARM, should be changed to iosArm32 or iosArm64
+    // For Linux, should be changed to e.g. linuxX64
+    // For MacOS, should be changed to e.g. macosX64
+    // For Windows, should be changed to e.g. mingwX64
+    ios()
+    sourceSets {
+        getByName("iosMain") {
+            dependsOn(getByName("darwinCommon"))
+        }
+        getByName("iosX64Main") {
+            dependsOn(getByName("iosMain"))
+        }
+        getByName("iosArm64Main") {
+            dependsOn(getByName("iosMain"))
+        }
+        getByName("iosTest") {
+        }
+    }
+}
+
+// Macos configuration
+kotlin {
+    // For ARM, should be changed to iosArm32 or iosArm64
+    // For Linux, should be changed to e.g. linuxX64
+    // For MacOS, should be changed to e.g. macosX64
+    // For Windows, should be changed to e.g. mingwX64
+    macosX64("macos") {}
+    sourceSets {
+        getByName("macosMain") {
+            dependsOn(getByName("darwinCommon"))
+        }
+        getByName("macosTest") {
+        }
+    }
+}
+
+// Needs running emulator
+// tasks.named("iosTest") {
+//    val device: String = project.findProperty("iosDevice")?.toString() ?: "iPhone 11 Pro Max"
+//    dependsOn(kotlin.targets.getByName<KotlinNativeTargetWithSimulatorTests>("ios").binaries.getTest("DEBUG").linkTaskName)
+//    group = JavaBasePlugin.VERIFICATION_GROUP
+//    description = "Runs tests for target 'ios' on an iOS simulator"
+//
+//    doLast {
+//        val binary = kotlin.targets.getByName<KotlinNativeTargetWithSimulatorTests>("ios").binaries.getTest("DEBUG").outputFile
+//        exec {
+//            commandLine("xcrun", "simctl", "spawn", device, binary.absolutePath)
+//        }
+//    }
+// }
+
+realmPublish {
+    pom {
+        name = "Sync"
+        description = "Sync code for Realm Kotlin. This artifact is not " +
+            "supposed to be consumed directly, but through " +
+            "'io.realm.kotlin:gradle-plugin:${Realm.version}' instead."
+    }
+}
+
+tasks.dokkaHtml.configure {
+    moduleName.set("Realm Kotlin Multiplatform SDK")
+    moduleVersion.set(Realm.version)
+    dokkaSourceSets {
+        configureEach {
+            moduleVersion.set(Realm.version)
+            reportUndocumented.set(true)
+            skipEmptyPackages.set(true)
+            perPackageOption {
+                matchingRegex.set("io\\.realm\\.internal\\.*")
+                suppress.set(true)
+            }
+            jdkVersion.set(8)
+        }
+        val commonMain by getting {
+            sourceRoot("../runtime-api/src/commonMain/kotlin")
+        }
+    }
+}
+
+tasks.register("uploadDokka") {
```

Comment:
I suspect we want to disable this in the base-library and only upload these from here. But maybe that is going to be done in another PR. If so, we should probably create an issue so we remember it.

Reply:
Sure, just created https://github.com/realm/realm-kotlin/issues/394